README.md

```diff
@@ -28,15 +28,86 @@ go-jsonschema \
+
+* Type property names will not change.
+* The `type` of properties will not change, except the allowed addition of `null`.
+* Type [title](https://json-schema.org/understanding-json-schema/reference/annotations) will not change.
+* Types will be not change to make validation more strict. Changes may occur if they make validation less strict. This applies to the following keywords. Examples are given, but they are not exhaustive.
+  * [minLength, maxLength](https://json-schema.org/understanding-json-schema/reference/string): `minLength` will not increase and `maxLength` will not decrease.
+  * [pattern](https://json-schema.org/understanding-json-schema/reference/string#regexp): pattern will not become stricter.
+  * [format](https://json-schema.org/understanding-json-schema/reference/string#built-in-formats): will not change.
+  * [multipleOf](https://json-schema.org/understanding-json-schema/reference/numeric#multiples): will not change.
+  * [minimum, exclusiveMinimum, maximum, exclusiveMaximum](https://json-schema.org/understanding-json-schema/reference/numeric#range): `minimum`, `exclusiveMinimum` will not increase; `maximum`, `exclusiveMaximum` will not decrease.
+  * [patternProperties](https://json-schema.org/understanding-json-schema/reference/object#patternProperties): will not expand scope to restrict additional properties.
+  * [additionalProperties](https://json-schema.org/understanding-json-schema/reference/object#additionalproperties): will not go from `true` to `false`.
+  * [propertyNames](https://json-schema.org/understanding-json-schema/reference/object#propertyNames): will not become stricter.
+  * [minProperties, maxProperties](https://json-schema.org/understanding-json-schema/reference/object#size): `minProperties` will not increase, `maxProperties` will not decrease.
+  * [required](https://json-schema.org/understanding-json-schema/reference/object#required): will not add additional entries.
+  * [contains, minContains, maxContains](https://json-schema.org/understanding-json-schema/reference/array#contains): will not become stricter.
+  * [minItems, maxItems](https://json-schema.org/understanding-json-schema/reference/array#length): `minItems` will not increase, `maxItems` will not decrease.
+  * [uniqueItems](https://json-schema.org/understanding-json-schema/reference/array#uniqueItems): will not go from `true` to `false`.
+  * [enum](https://json-schema.org/understanding-json-schema/reference/enum): will not remove entries.
+  * [const](https://json-schema.org/understanding-json-schema/reference/const): will not change.
+* No existing type will be deleted.
+* No type property will be deleted.
+
+The following additive changes are allowed:
+
+* Adding of new properties to existing types.
+* Adding new types.
+* Changes that make property validation less strict. See above for examples.
+* Removing a property from `required`.
```

Comment:
Should marking a feature deprecated be captured here? Maybe we can capture this with some form of annotation

Reply:
JSON schema has a concept called [annotations](https://json-schema.org/understanding-json-schema/reference/annotations), with a specific `deprecated` keyword: 

> JSON Schema includes a few [keywords](https://json-schema.org/learn/glossary#keyword), that aren't strictly used for validation, but are used to describe parts of a schema.

The key bit here is that annotations are part of validation. Though (and maybe this is what you're getting at) they may be used in code generation, similar to the `title` annotation.

I'm happy to include a note about deprecated here. I think we should be free to add / remove deprecated freely. I.e. it would be acceptable to add it, then later remove it.

Reply:
I ended up adding bullet points for `description` and `deprecated` annotations.